Gold: 2026-fix
--- a/Source/WebCore/platform/graphics/vulkan/VulkanUtilities.cpp
+++ b/Source/WebCore/platform/graphics/vulkan/VulkanUtilities.cpp
@@ -200,6 +200,7 @@
     bool shouldInstallDebugMessenger = false;
 #endif // VK_EXT_debug_utils
 
+#if !LOG_DISABLED || !RELEASE_LOG_DISABLED
     if (LOG_CHANNEL(Vulkan).state != WTFLogChannelState::Off) [[unlikely]] {
         static constexpr auto validationLayerName = "VK_LAYER_KHRONOS_validation";
         if (Instance::hasLayer(validationLayerName)) {
@@ -230,6 +231,7 @@
         }
         RELEASE_LOG_DEBUG(Vulkan, "Requesting %zu extensions%s", extensionNames.size(), formattedExtensionList.toString().utf8().data());
     }
+#endif // !LOG_DISABLED || !RELEASE_LOG_DISABLED
 
 #if USE(GLIB)
     const String applicationName = makeString(getApplicationID(), '.', processTypeDescription(processType()));
--- a/Source/WebCore/platform/graphics/vulkan/VulkanTypes.cpp
+++ b/Source/WebCore/platform/graphics/vulkan/VulkanTypes.cpp
@@ -504,6 +504,7 @@
 {
     ASSERT(data->pMessage);
 
+#if !LOG_DISABLED || !RELEASE_LOG_DISABLED
     const WTFLogLevel logLevel = [messageSeverity] {
         switch (messageSeverity) {
         case VK_DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT:
@@ -533,6 +534,11 @@
         RELEASE_LOG_WITH_LEVEL(Vulkan, logLevel, "[%s: %s] %s", data->pMessageIdName, messageTypeString.toString().utf8().data(), data->pMessage);
     else
         RELEASE_LOG_WITH_LEVEL(Vulkan, logLevel, "[%s] %s", messageTypeString.toString().utf8().data(), data->pMessage);
+#else
+    (void)messageSeverity;
+    (void)messageTypes;
+    (void)data;
+#endif // !LOG_DISABLED || !RELEASE_LOG_DISABLED
 
     return VK_FALSE;
 }
@@ -546,12 +552,14 @@
     createInfo->pfnUserCallback = debugUtilsMessengerHandleMessage;
     createInfo->messageType = VK_DEBUG_UTILS_MESSAGE_TYPE_GENERAL_BIT_EXT | VK_DEBUG_UTILS_MESSAGE_TYPE_VALIDATION_BIT_EXT | VK_DEBUG_UTILS_MESSAGE_TYPE_VALIDATION_BIT_EXT;
     createInfo->messageSeverity = VK_DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT;
+#if !LOG_DISABLED || !RELEASE_LOG_DISABLED
     if (LOG_CHANNEL(Vulkan).level >= WTFLogLevel::Info)
         createInfo->messageSeverity |= VK_DEBUG_UTILS_MESSAGE_SEVERITY_INFO_BIT_EXT;
     if (LOG_CHANNEL(Vulkan).level >= WTFLogLevel::Warning)
         createInfo->messageSeverity |= VK_DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT;
     if (LOG_CHANNEL(Vulkan).level >= WTFLogLevel::Debug)
         createInfo->messageSeverity |= VK_DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT;
+#endif // !LOG_DISABLED || !RELEASE_LOG_DISABLED
 
     return vkCreateDebugUtilsMessengerEXT(ptr(), createInfo.ptr(), nullptr, &m_debugMessenger);
 }
