From 62ae6417d02a42c2ce9dee77922cb7a90e89a375 Mon Sep 17 00:00:00 2001
From: noname
Date: Sun, 2 Aug 2026 17:53:58 +0200
Subject: [PATCH 2/2] Add multiple soft reset functions

---
 drivers/net/phy/phy_device.c | 32 +++++++++++++++++++++++++-------
 1 file changed, 25 insertions(+), 7 deletions(-)

diff --git a/drivers/net/phy/phy_device.c b/drivers/net/phy/phy_device.c
index 1d7881250364..4f5a3b32fd2a 100644
--- a/drivers/net/phy/phy_device.c
+++ b/drivers/net/phy/phy_device.c
@@ -1019,6 +1019,20 @@ struct phy_device *get_phy_device(struct mii_bus *bus, int addr, bool is_c45)
 }
 EXPORT_SYMBOL(get_phy_device);
 
+static int phy_device_soft_reset(struct phy_device *phydev)
+{ 
+	int ret = 0;
+	
+	if (phydev->drv->soft_reset) {
+		ret = phydev->drv->soft_reset(phydev);
+		/* see comment in genphy_soft_reset for an explanation */
+		if (!ret)
+			phydev->suspended = 0;
+	}
+	
+	return ret;
+}
+
 /**
  * phy_device_register - Register the phy device on the MDIO bus
  * @phydev: phy_device structure to be added to the MDIO bus
@@ -1033,6 +1047,7 @@ int phy_device_register(struct phy_device *phydev)
 
 	/* Deassert the reset signal */
 	phy_device_reset(phydev, 0);
+	phy_device_soft_reset(phydev);
 
 	/* Run all of the fixups for this PHY */
 	err = phy_scan_fixups(phydev);
@@ -1052,6 +1067,7 @@ int phy_device_register(struct phy_device *phydev)
  out:
 	/* Assert the reset signal */
 	phy_device_reset(phydev, 1);
+	phy_device_soft_reset(phydev);
 
 	mdiobus_unregister_device(&phydev->mdio);
 	return err;
@@ -1075,6 +1091,7 @@ void phy_device_remove(struct phy_device *phydev)
 
 	/* Assert the reset signal */
 	phy_device_reset(phydev, 1);
+	phy_device_soft_reset(phydev);
 
 	mdiobus_unregister_device(&phydev->mdio);
 }
@@ -1263,12 +1280,7 @@ int phy_init_hw(struct phy_device *phydev)
 	if (!phydev->drv)
 		return 0;
 
-	if (phydev->drv->soft_reset) {
-		ret = phydev->drv->soft_reset(phydev);
-		/* see comment in genphy_soft_reset for an explanation */
-		if (!ret)
-			phydev->suspended = 0;
-	}
+	ret = phy_device_soft_reset(phydev);
 
 	if (ret < 0)
 		return ret;
@@ -1877,6 +1889,7 @@ void phy_detach(struct phy_device *phydev)
 
 	/* Assert the reset signal */
 	phy_device_reset(phydev, 1);
+	phy_device_soft_reset(phydev);
 
 	/*
 	 * The phydev might go away on the put_device() below, so avoid
@@ -2001,6 +2014,7 @@ int phy_reset_after_clk_enable(struct phy_device *phydev)
 	if (phydev->drv->flags & PHY_RST_AFTER_CLK_EN) {
 		phy_device_reset(phydev, 1);
 		phy_device_reset(phydev, 0);
+		phy_device_soft_reset(phydev);
 		return 1;
 	}
 
@@ -3319,6 +3333,7 @@ static int phy_probe(struct device *dev)
 
 	/* Deassert the reset signal */
 	phy_device_reset(phydev, 0);
+	phy_device_soft_reset(phydev);
 
 	if (phydev->drv->probe) {
 		err = phydev->drv->probe(phydev);
@@ -3415,8 +3430,10 @@ static int phy_probe(struct device *dev)
 
 out:
 	/* Re-assert the reset signal on error */
-	if (err)
+	if (err){
 		phy_device_reset(phydev, 1);
+		phy_device_soft_reset(phydev);
+	}
 
 	return err;
 }
@@ -3441,6 +3458,7 @@ static int phy_remove(struct device *dev)
 
 	/* Assert the reset signal */
 	phy_device_reset(phydev, 1);
+	phy_device_soft_reset(phydev);
 
 	phydev->drv = NULL;
 
-- 
2.43.0

